src/miral/application_selector.h

```diff
@@ -90,6 +91,7 @@ class ApplicationSelector
 
     /// The application that is currently selected.
     Window selected;
+    MirWindowState restore_state = mir_window_state_unknown;
```

Comment:
`restore_state` seems to be a good candidate for `std::optional<MirWindowState>` rather than an in-band sentinal?

Reply:
Agreed